aws/credentials/process_provider.go

```diff
@@ -0,0 +1,228 @@
+	config, err := ini.Load(filename)
+	if err != nil {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", "failed to load shared config file", err)
+	}
+
+	var iniProfile *ini.Section
+	if len(profile) == 0 || profile == "default" {
+		iniProfile, err = config.GetSection("default")
+	} else {
+		iniProfile, err = config.GetSection("profile " + profile)
+	}
+	if err != nil {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", "failed to get profile", err)
+	}
+
+	process, err := iniProfile.GetKey("credential_process")
+	if err != nil {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", "No credential_process specified", err)
+	}
+	return p.credentialProcess(process.String())
+}
+
+type credentialProcessResponse struct {
+	Version         int
+	AccessKeyID     string `json:"AccessKeyId"`
+	SecretAccessKey string
+	SessionToken    string
+	Expiration      string
+}
+
+func executeCredentialProcess(process string) ([]byte, error) {
+	processArgs := strings.Split(process, " ")
+	var command string
+	var cmdArgs []string
+
+	if len(processArgs) == 1 {
+		command = processArgs[0]
+	} else if len(processArgs) > 1 {
+		command = processArgs[0]
+		cmdArgs = processArgs[1:]
+	}
+
+	// TODO: Check for executability?
+	_, err := os.Stat(command)
+	if os.IsNotExist(err) {
+		return nil, awserr.New("ProcessProviderLoad", "credential_process "+command+" not found", err)
+	} else if err != nil {
+		return nil, awserr.New("ProcessProviderLoad", "failed to open credential_process "+command, err)
+	}
+
+	// Execute command
+	cmd := exec.Command(command, cmdArgs...)
+	cmd.Env = os.Environ()
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, awserr.New("ProcessProviderLoad", "Error executing credential_process", err)
+	}
+	return out, nil
+}
+
+func (p *ProcessProvider) credentialProcess(process string) (Value, error) {
+	out, err := p.executionFunc(process)
+	if err != nil {
+		return Value{ProviderName: ProcessProviderName}, err
+	}
+
+	// Serialize and validate response
+	resp := &credentialProcessResponse{}
+	err = json.Unmarshal(out, resp)
+	if err != nil {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", "Error parsing credential_process output: "+string(out), err)
+	}
+	if resp.Version != 1 {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", " Version in credential_process output is not 1", err)
+	}
+	if len(resp.AccessKeyID) == 0 {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", " Missing required AccessKeyId in credential_process output", err)
+	}
+	if len(resp.SecretAccessKey) == 0 {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", " Missing required SecretAccessKey in credential_process output", err)
+	}
+
+	// Handle expiration
+	if len(resp.Expiration) > 0 {
```

Comment:
Using a pointer for `resp.Expiration` would simplify receiving the time value. This is done for the SDK's [endpoint credential provider](https://github.com/aws/aws-sdk-go/blob/master/aws/credentials/endpointcreds/provider.go#L134)